examples/low-level-api/common.rs

```diff
@@ -77,14 +76,14 @@ pub async fn add_new_key(
   // Add #newKey to the document
   let new_key: KeyPair = KeyPair::new_ed25519()?;
   let method: IotaVerificationMethod =
-    IotaVerificationMethod::from_did(updated_doc.did().clone(), new_key.type_(), new_key.public(), "newKey")?;
+    IotaVerificationMethod::from_did(updated_doc.id().clone(), new_key.type_(), new_key.public(), "newKey")?;
```

Comment:
Discussion: I would personally prefer having a getter named `did` return the did, because it seems more consistent. This applies to `IotaDocument` and `CoreDocument`. For verification methods, maybe `did_url` or `url` would make sense. Do we have `id` because there is a preference for that, or is it just to avoid name clashing with `TangleRef::did`? If I recall correctly, didn't we want to remove `TangleRef`? If so, we could make the change to `did`, if there is a general preference.

Reply:
I believe it's called `id` because that's the name of the field in the DID document proper, according to the W3C specification: https://www.w3.org/TR/did-core/#identifiers

`TangleRef` is still used in a few places, I think it could be simplified but not removed since `message_ext` uses it to abstract decoding DID messages (integration or diff messages).

Reply:
Would you prefer the `TangleRef` refactor to be done here or can it wait for a different PR?

Reply:
Looked into removing `TangleRef` briefly again and it looks like it's used in `MessageIndex`, `sort_by_milestone`, and `parse_data`. All but one of the `TangleRef` methods are used, with the unused method being `set_previous_message_id`. So I don't think it is feasible to remove or even simplify.

Reply:
Alright, thanks very much for looking into it. At least we have consistency with the specification regarding `id`, which is nice.